.circleci/config.yml

```diff
@@ -10,22 +10,31 @@ jobs:
       - run:
           name: Check Twitter handles
           command: |
-            gem i twitter --no-post-install-message --no-suggestions --minimal-deps --no-verbose -N -q
-            bash twitter.sh
+            git --no-pager diff origin/master..HEAD ../_data | grep "^+[[:space:]]" | grep twitter | cut -c15-
+            [[ $? -z ]] && { gem i twitter --no-post-install-message --no-suggestions --minimal-deps --no-verbose -N -q; ruby twitter.rb }
```

Comment:
Are you sure about `[[ $? -z ]]`? That doesn't look valid to me.

Reply:
Checks the exit code of the last executed command. Turns true if it’s 0.

Reply:
> % true; [[ $? -z ]] && echo "that is true"
> -bash: conditional binary operator expected
> -bash: syntax error near `-z'
> % echo $BASH_VERSION
> 4.2.46(1)-release

Do you mean `[[ $? -eq 0 ]] && ...`?

Reply:
Whoops!
Fixed :grin:

Reply:
Okay this is weird. CircleCI on my fork keeps failing at this line :thinking:

Reply:
My local code is:
```bash
[[ -z "$(git --no-pager diff origin/master..HEAD ../_data | grep "^+[[:space:]]" | grep twitter | cut -c15-)" ]] || { sudo gem i twitter --no-post-install-message --no-suggestions --minimal-deps --no-verbose -N -q; ruby twitter.rb; } && echo "No Twitter handles found. Skipping checks.";
```
Which echos "No Twitter handles found. Skipping checks." (returns 0) on the git diff returning null.
If the git diff returns anything it installs the gem and runs `./tests/Twitter.rb`.

Reply:
That's the short circuit evaluation of boolean expressions in bash:
```
% a="not empty"
% [[ -z "$a" ]] || echo "part 1" && echo "part 2"
part 1
part 2
% [[ -n "$a" ]] && echo "part 2" || echo "part 1"
part 2
```
```
% a=""
% [[ -z "$a" ]] || echo "part 1" && echo "part 2"
part 2
% [[ -n "$a" ]] && echo "part 2" || echo "part 1"
part 1
```

`&&` has a higher precedence than `||`.

Reply:
Hmm but it should still exit 0 regardless of which part is run.